### PR TITLE
feat: submit private links

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes

Will be able to test once the [PR for the relevant API](https://github.com/dailydotdev/daily-api/pull/1203) is merged.

- Introduced a new component for the Squad Post Bar.
- Made the Modal to add a Squad post accept a link and render conditionally.
- Refactored the Modal to use a different endpoint if link was included in the parameters.
- Refactored the Squad header bar.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-WT-1132 #done

### CODE REVIEW

File: packages/shared/src/components/squads/SharePostBar.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  onNewSquadPost,
+}: SharePostBarProps): ReactElement {
+  const [url, setUrl] = useState('');
+  const { mutateAsync: getPost } = useMutation(getPostByUrl, {
+    onSuccess: (post) => {
+      onNewSquadPost({ post });
+    },
+    onError: () => {
+      onNewSquadPost({ url });
+    },
```
Comment (rebelchris) [inline packages/shared/src/components/squads/SharePostBar.tsx:31]: Depending on the error we might want to show the error already here.
So example being invalid URL or deleted post.

File: packages/shared/src/components/modals/PostToSquadModal.tsx [hunk 2]
```diff
@@ -91,8 +107,12 @@ function PostToSquadModal({
       steps={post ? undefined : modalSteps}
     >
       <Modal.Header title="Share post" kind={ModalHeaderKind.Tertiary} />
-      {post ? (
-        <SquadComment form={form} onSubmit={onSubmit} isLoading={isLoading} />
+      {shouldSkipHistory ? (
```
Comment (rebelchris) [inline packages/shared/src/components/modals/PostToSquadModal.tsx:110]: This seems to cause a flicker locally atleast.

File: packages/shared/src/components/squads/SharePostBar.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        size="large"
+        nativeLazyLoading
+      />
+      <input
+        type="url"
+        name="share-post-bar"
+        className="flex-1 pl-1 outline-none bg-theme-bg-transparent text-theme-label-primary focus:placeholder-theme-label-quaternary hover:placeholder-theme-label-primary typo-callout"
+        placeholder="Enter link to share"
+        onInput={(e) => setUrl(e.currentTarget.value)}
+        required
```
Comment (rebelchris) [inline packages/shared/src/components/squads/SharePostBar.tsx:61]: Should be non-required as user can simply press "Post" as empty field.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice start!
Some small comments mainly for styling now.

Missing the other header adjustments:
Screenshot 2023-03-09 at 09 48 05

File: packages/shared/src/components/squads/SharePostBar.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/SharePostBar.tsx:31]: Right. Let me catch it.

File: packages/shared/src/components/squads/SharePostBar.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/SharePostBar.tsx:61]: Good catch!

**Comment by sshanzel** [comment]
> @sshanzel I also noticed the getPostByUrl seems to throw an error underwater "could not find post" think we need to catch it and proceed as "ok" in those cases.

It only shows as prompt in development, but that is actually caught by the `useMutation` where the `onError` is triggered.

File: packages/shared/src/components/modals/PostToSquadModal.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/modals/PostToSquadModal.tsx:110]: Should be okay now. There were two areas triggering the show of the modal.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Some generic remarks:

- Can just click on "Post" (should be possible)
- It validates URL in the header bar already, (browser validation)
- If I remove the URL in the second screen nothing happens (should show that it's required)
- If I paste new URL, then remove it and paste existing article it should show that article on the comment modal.

Let me know if you want me to show you certain things.

**Comment by sshanzel** [comment]
> Let me know if you want me to show you certain things.

I will finish first the concern in the BE then I will get back to this.

**Comment by sshanzel** [comment]
> Can just click on "Post" (should be possible)

Fixed! Thank you!

> It validates URL in the header bar already, (browser validation)

Do we not want to use the browser's native validation? I think it is very useful.

> If I remove the URL in the second screen nothing happens (should show that it's required)

It turns the left border to red as part of being required. Do you want to add an error text below?

image

> If I paste new URL, then remove it and paste existing article it should show that article on the comment modal.

Have finished this by will do more local tests tomorrow first thing.

**Comment by sshanzel** [comment]
@rebelchris private links submission should now work as intended. I have also included a hint for when there is an error.

**Comment by sshanzel** [comment]
Currently deployed at preview2. Quick link: https://preview2.app.daily.dev/squads/dailydotdev

File: packages/shared/src/components/squads/SquadHeaderBar.tsx [hunk 4]
```diff
@@ -39,20 +51,28 @@ export function SquadHeaderBar({
...
-          onClick={onMenuClick}
-          aria-label="Squad options"
+          tag="a"
+          target="_blank"
+          rel="noopener noreferrer"
+          href={`${squadFeedback}#user_id=${user.id}&squad_id=${squad.id}`}
+          className="btn-secondary"
+          icon={<FeedbackIcon aria-label="squad-feedback-icon" />}
+          aria-label="squad-feedback"
         />
```
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadHeaderBar.tsx:66]: We should probably wrap this in a tooltip now as it's not super descriptive what the button does.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Wow, almost there!

Initial state has margin below url with no error:
Screenshot 2023-03-10 at 13 12 42

Error is not red:
Screenshot 2023-03-10 at 13 12 50

File: packages/shared/src/components/squads/SquadHeaderBar.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/SquadHeaderBar.tsx:66]: Totally agree with that.

**Comment by sshanzel** [comment]
> Initial state has margin below url with no error:

That was actually intended, built-in with our input field, to avoid content shift on responsive error message. Probably I should reduce the bottom padding of the container (if there was any).

**Comment by sshanzel** [comment]
> @sshanzel The only issue I still see is the text no always being red.
It seems to only happen on re-focus.
> Maybe just always make it red, regardless of state?
It's an error so won't ever show differently right.

That is actually part of our design guideline. The red text would only happen on validation.

image

**Review by rebelchris — APPROVED** [review]
Maybe also ask ido to review this one.
From that point if looks complete though.

Thanks for the guideline, wasn't aware of that part.

**Comment by sshanzel** [comment]
@rebelchris thank you for the awesome review! 🙇 

@idoshamun as Chris suggested, kindly review if you have some spare time. This is currently deployed at preview2.

Quick link: https://preview2.app.daily.dev/squads/dailydotdev

**Comment by rebelchris** [comment]
@idoshamun One concern I have is that `postByUrl` now throws an error, not sure it's bad but could potentially be improved, so maybe something you can evaluate if it's ok for now?

**Comment by idoshamun** [comment]
@rebelchris @sshanzel what are the cases the user can see an error?

**Comment by sshanzel** [comment]
> @rebelchris @sshanzel what are the cases the user can see an error?

When the URL was invalid or when the URL is empty. Both cases happen after when you focus out which triggers form validation / submitting the form.

@rebelchris let me know if you have seen any responsive issues with the current updates.

**Pull request merged**